samples/encode-decode-worker/js/stream_worker.js

```diff
@@ -0,0 +1,260 @@
+   DecodeVideoStream(self) {
+     return new TransformStream({
+       start(controller) {
+         this.decoder = decoder = new VideoDecoder({
+           output: frame => controller.enqueue(frame),
+           error: (e) => {
+              self.postMessage({severity: 'fatal', text: `Init Decoder error: ${e.message}`});
+           }
+         });
+       },
+       transform(chunk, controller) {
+         if (this.decoder.state != "closed") {
+           if (chunk.type == "config") {
+              let config = JSON.parse(chunk.config);
+              VideoDecoder.isConfigSupported(config).then((decoderSupport) => {
+                if(decoderSupport.supported) {
+                  this.decoder.configure(decoderSupport.config);
+                  self.postMessage({text: 'Decoder successfully configured:\n' + JSON.stringify(decoderSupport.config)});
+                } else {
+                self.postMessage({severity: 'fatal', text: 'Config not supported:\n' + JSON.stringify(decoderSupport.config)});
+                }
+              })
+              .catch((e) => {
+                 self.postMessage({severity: 'fatal', text: `Configuration error: ${e.message}`});
+              })
+           } else {
+             try {
+               const queue = this.decoder.decodeQueueSize;
+               decqueue_update(queue);
+               this.decoder.decode(chunk);
+             } catch (e) {
+               self.postMessage({severity: 'fatal', text: 'Derror size: ' + chunk.byteLength + ' seq: ' + chunk.seqNo + ' kf: ' + chunk.keyframeIndex + ' delta: ' + chunk.deltaframeIndex + ' dur: ' + chunk.duration + ' ts: ' + chunk.timestamp + ' ssrc: ' + chunk.ssrc + ' pt: ' + chunk.pt + ' tid: ' + chunk.temporalLayerId + ' type: ' + chunk.type});
+               self.postMessage({severity: 'fatal', text: `Catch Decode error: ${e.message}`});
+             }
+           }
+         }
+       }
+     });
+   }
+
+   EncodeVideoStream(self, config) {
+     return new TransformStream({
+       start(controller) {
+         this.frameCounter = 0;
+         this.seqNo = 0;
+         this.keyframeIndex = 0;
+         this.deltaframeIndex = 0;
+         this.pending_outputs = 0;
+         this.encoder = encoder = new VideoEncoder({
+           output: (chunk, cfg) => {
+             if (cfg.decoderConfig) {
+               const decoderConfig = JSON.stringify(cfg.decoderConfig);
+               self.postMessage({text: 'Configuration: ' + decoderConfig});
+               const configChunk =
+               {
+                  type: "config",
+                  seqNo: this.seqNo,
+                  keyframeIndex: this.keyframeIndex,
+                  deltaframeIndex: this.deltaframeIndex,
+                  timestamp: 0,
+                  pt: 0,
+                  config: decoderConfig 
+               };
+               controller.enqueue(configChunk); 
+             } 
+             chunk.temporalLayerId = 0;
+             if (cfg.svc.temporalLayerId) {
```

Comment:
When scalability_mode is configured to L1T1, the UA might not set cfg.svc in its outout. Do you think the UA is not following spec, or you think it better we check if cfg.svc is defined or not?

Reply:
Before inserting this check, I had encountered a situation where `cfg.svc` was not defined, so the lines 178 and 179 were inserted to cover that situation.

Reply:
with your current change it will report temporalLayerId undefined on L1T1, so you should fist check cfg.svc, not cfg.svc.temporalLayerId.

Reply:
@djuffin for comment

Reply:
The spec says:

---
If encoderConfig.[scalabilityMode](https://www.w3.org/TR/webcodecs/#dom-videoencoderconfig-scalabilitymode) describes multiple [temporal layers](https://www.w3.org/TR/webcodecs/#temporal-layer):

Let svc be a new [SvcOutputMetadata](https://www.w3.org/TR/webcodecs/#dictdef-svcoutputmetadata) instance.

Let temporal_layer_id be the zero-based index describing the temporal layer for output.

Assign temporal_layer_id to svc.[temporalLayerId](https://www.w3.org/TR/webcodecs/#dom-svcoutputmetadata-temporallayerid).

Assign svc to chunkMetadata.[svc](https://www.w3.org/TR/webcodecs/#dom-encodedvideochunkmetadata-svc).

---

L1T1 doesn't describe multiple temporal layers, so `svc` entry doesn't have to be there.

Reply:
Fixed.